services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DelegatingDeviceManagementHttpEndpoint.java

```diff
@@ -93,21 +93,25 @@ public void addRoutes(final Router router) {
 
         // CREATE device
         router.post(pathWithTenantAndDeviceId)
+                .handler(this::verifyDeviceId)
```

Comment:
I think it would be better to do this check within span, so that we have a proper tracing/logging for the errors.